File: TownOfSalemCalculator/src/townofsalemcalculator/Conditions/Concrete/NightMessages/Nursed.java
package townofsalemcalculator.Conditions.Concrete.NightMessages;

import java.util.Arrays;
import java.util.List;
import static townofsalemcalculator.Role.Doctor;
import townofsalemcalculator.Conditions.RoleGroup.SingleRoleGroup;

/**
 * The Concrete Condition that you were healed
 * @author Multifacio
 * @version 1.1
 * @since 2017-1-28
 */
public class Nursed extends NightMessage {    
    public Nursed() {
        super(new SingleRoleGroup(Doctor));
    }
    
    @Override
    public String getCondition() {
        return "You were attacked but someone nursed you back to health";
    }

    @Override
    public List<String> keyWords() {
        List<String> keyWords = Arrays.asList(new String[]{"Attacked", "Nursed", "Back", "Health", "Healed", "You"});
        keyWords.addAll(Doctor.getKeyWords());
        return keyWords;
    }
}

File: src/Conditions/Concrete/Effects/Nursed.py
from src.Concepts.Role import Role
from src.Conditions.Concrete.ExistenceCondition import ExistenceCondition
from typing import FrozenSet

class Nursed(ExistenceCondition):
    """ The Nursed condition is used when during the night you were attacked but also healed by a Doctor, i.e.
    you received the message: 'You were attacked but someone nursed you back to health' """

    def __init__(self, amnesiacRemembered: FrozenSet[Role] = None):
        super().__init__(frozenset({Role.DOCTOR}), amnesiacRemembered)